lib/matplotlib/figure.py

```diff
@@ -1327,130 +1328,18 @@ def legend(self, *args, **kwargs):
-
-        framealpha : None or float
-            Control the alpha transparency of the legend's background.
-            Default is *None* which will take the value from the
-            ``legend.framealpha`` :data:`rcParam<matplotlib.rcParams>`.
-
-        facecolor : None or "inherit" or a color spec
-            Control the legend's background color.
-            Default is *None* which will take the value from the
-            ``legend.facecolor`` :data:`rcParam<matplotlib.rcParams>`.
-            If ``"inherit"``, it will take the ``axes.facecolor``
-            :data:`rcParam<matplotlib.rcParams>`.
-
-        edgecolor : None or "inherit" or a color spec
-            Control the legend's background patch edge color.
-            Default is *None* which will take the value from the
-            ``legend.edgecolor`` :data:`rcParam<matplotlib.rcParams>`.
-            If ``"inherit"``, it will take the ``axes.edgecolor``
-            :data:`rcParam<matplotlib.rcParams>`.
-
-        ncol : integer
-            Number of columns. Default is 1.
-
-        mode : "expand" or None
-            If mode is "expand", the legend will be horizontally expanded
-            to fill the axes area (or *bbox_to_anchor*)
-
-        title : string
-            The legend title
-
-        borderpad : float or None
-            The fractional whitespace inside the legend border, measured in
-            font-size units.
-            Default is *None* which will take the value from the
-            ``legend.borderpad`` :data:`rcParam<matplotlib.rcParams>`.
-
-        labelspacing : float or None
-            The vertical space between the legend entries, measured in
-            font-size units.
-            Default is *None* which will take the value from the
-            ``legend.labelspacing`` :data:`rcParam<matplotlib.rcParams>`.
-
-        handlelength : float or None
-            The length of the legend handles, measured in font-size units.
-            Default is *None* which will take the value from the
-            ``legend.handlelength`` :data:`rcParam<matplotlib.rcParams>`.
-
-        handletextpad : float or None
-            The padding between the legend handle and text, measured in
-            font-size units.
-            Default is *None* which will take the value from the
-            ``legend.handletextpad`` :data:`rcParam<matplotlib.rcParams>`.
-
-        borderaxespad : float or None
-            The padding between the axes and legend border, measured in
-            font-size units.
-            Default is *None* which will take the value from the
-            ``legend.borderaxespad`` :data:`rcParam<matplotlib.rcParams>`.
-
-        columnspacing : float or None
-            The spacing between columns, measured in font-size units.
-            Default is *None* which will take the value from the
-            ``legend.columnspacing`` :data:`rcParam<matplotlib.rcParams>`.
+
+        %(_legend_kw_doc)s
```

Comment:
At some point, it had been decided to stop using docstring interpolation. They often break and are hard to maintain.

I don't know where that is.

Reply:
That is unfortunate, but I'm more than happy to do this the "right" way.   

Is there a better way to not write out the same doc string three times?  The fundamental issue here was that `figure.legend`, `axes.legend` and `Legend.__init__` all have the same kwargs, and *all three* had a different version of the documentation for them.  So that is also hard to maintain, and was definitely broken.   There has to be a way to share parts of doc strings, and I assumed this was it based on the existing code base.  

As for discoverability, most IDEs will let you search for a string in a project.  `_legend_kw_doc` is in `legend.py`.  

However, if there is a better way to specify the doc string so its easy to maintain and share parts of the string, I'm more than happy to switch!   Thanks.

Reply:
No, there isn't an easy way not to write the docstring three times, apart from docstring interpolation.

For very "common" options, the idea was to link back to one place containing the information (linestyle options, colors, markers, etc). 
In that particular case, I would just copy paste the kwargs.

The question is whether you want to favor consistency over simplicity. I favor simplicity, as I believe documentation should be easily editable by anyone. Some contributors believe that consistency is more important.

Reply:
I think the interpolation string is editable by anyone, though I'll readily admit I had considerable trouble getting the indentation correct!   But I didn't find it particularly complicated.  

Looking at the code base, I think the historic problem is that people used names like `Axes` for the documentation string, and that is truly undiscoverable.  

I think consistency is pretty important.  If we add or subtract a kwarg from `legend` in the future, someone has to remember to change three doc strings, or have the documentation be wrong in the future.  The whole mess with this PR is that the docs were not consistent and the kwargs were not handled consistently, so it lead to user complaints.  

I'm happy to do as I'm told.  But I think having a way to include common docstrings is pretty powerful.  I'd certainly agree, however, that they should have a strong naming convention so they are easily searchable.  i.e. maybe `module_function_(*_)_docstring` (where the (*_) is an optional descriptor).

Reply:
Experiences showed over and over again that our docstring hacking makes Matplotlib's documentation extremely hard to edit by beginners in contributing to Matplotlib, yet fairly advanced python users.

FYI, that includes core-developers not noticing that our html documentation was eating part of our docstrings for years.